advice_animal/runner.py

```diff
@@ -11,10 +15,162 @@
 LOG = logging.getLogger(__name__)
 
 
+def run_cmd(cmd: list[Union[str, Path]], check: bool = True) -> Tuple[str, int]:
+    LOG.info("Run %s in %s", cmd, os.getcwd())
+    proc = subprocess.run(cmd, encoding="utf-8", capture_output=True, check=check)
+    LOG.debug("Ran %s -> %s", cmd, proc.returncode)
+    LOG.debug("Stdout: %s", proc.stdout)
+    return proc.stdout, proc.returncode
+
+
+@dataclass
+class Result:
+    advice_name: str
+    success: bool
+    project: str = ""
```

Comment:
imagine a structure like

```
core/
plugins/foo/
plugins/bar/
```

are you intending this to be the whole path, or the last path component, or something else?